waspc/cli/src/Wasp/Cli/Question.hs

```diff
@@ -0,0 +1,86 @@
+module Wasp.Cli.Question
```

Comment:
Interesting, I like how you isolated this!

I think you wanted to name the module `Query`, not `Question`? Although even `Query` is a bit vague, but at least it seems to be more in line with the naming you use for the functions in the module.

So you created your module for queries toward user, that makes sense. I have a feeling that some advanced haskeller would probably make this a monad or applicative or something hehe :D, but we don't have to overdo it for now, I think this is a good start.

I did find it somewhat hard to read though, maybe we can clean it up somewhat?

Some ideas:
1. It is unusual to see functions that are not verbs. Can work, but a bit confusing.
2. `type Question = String` and `type Answer = String` could already help somewhat.
3. Make sure stuff is more in context / more readable by providing better names, better order / structure of functions, ... .

Reply:
I've stolen the naming from the `summoner` https://github.com/kowainik/summoner/blob/main/summoner-cli/src/Summoner/Question.hs

I'll clean a bit what I think it makes sense.

Reply:
Well maybe they don't have the best naming :D. It is good to use them for inspiration, but we should be able to support it logically.

Query being a verb and noun -> ah sure, it is a verb also, but for me what made it sound like a noun in this context was the addition of `WithOptions` and `NotNull`. But I guess it depends on perspective hm. Still, not very clear from the name what the function does.

Reply:
I hope it's better with `Interactive` module name and `askForInput` and `askToChoose`